ExpressionUtils/Evaluating/CachedExpressionCompiler.cs

```diff
@@ -82,7 +83,29 @@ private static object getDefaultValue(Type t) {
 		/// Use for testing only.
 		/// </remarks>
 		internal bool IsCached(LambdaExpression lambda) {
-			return delegates.ContainsKey(lambda);
+			return delegates.ContainsKey((lambda.Body, lambda.Parameters));
+		}
+
+		/// <summary>
+		/// Previously as a key for <see cref="delegates"/> we were using <see cref="LambdaExpression"/> objects and <see cref="ExpressionComparing.StructuralComparer"/> as a comparer.
+		/// But that required us to make calls to <see cref="Expression.Lambda(Expression, ParameterExpression[])"/> which contains global lock inside, and that started to become a problem.

```

Comment:
The global lock stems from .net's caching mechanism right? Maybe mention that here?

Reply:
I dont think its a good idea to do so. Since its part of dotnet that can change. For example they changed implementation of that method in .net 9 (still seems to contain lock). So it seems to me that the best way is to just say that it was expensive due to global lock (this way its enough info to understand why this change was needed) and to not include additional details.

Reply:
I'm fine either way. Just wanted to confirm that the lock is something we cannot avoid (and it is not caused by ourselves, e.g. by the way we create the expressions)